crates/router/src/services/authentication.rs

```diff
@@ -1378,9 +1380,10 @@ where
         };
         Ok((
             auth.clone(),
-            AuthenticationType::MerchantJwt {
+            AuthenticationType::MerchantJwtWithProfileId {
                 merchant_id: auth.merchant_account.get_id().clone(),
-                user_id: Some(payload.user_id),
+                profile_id: auth.profile_id.clone(),
```

Comment:
Getting
```
borrow of moved value: `payload.profile_id`
move occurs because `payload.profile_id` has type `std::option::Option<common_utils::id_type::ProfileId>`, which does not implement the `Copy` trait
```
on `profile_id: payload.profile_id` because of
```
let auth = AuthenticationData {
            merchant_account: merchant,
            key_store,
            profile_id: payload.profile_id,
        };
```
above it.

What is the correct way to deal with this?

Reply:
You have to clone the above usage of `payload.profile_id`.